File: android/src/main/java/com/tianxin/flutter_aiui/FlutterAiuiPlugin.java
package com.tianxin.flutter_aiui;

import android.app.Activity;
import android.content.Context;

import androidx.annotation.NonNull;

import java.util.Objects;

import io.flutter.embedding.engine.plugins.FlutterPlugin;
import io.flutter.embedding.engine.plugins.activity.ActivityAware;
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;
import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;

/**
 * FlutterAiuiPlugin
 */
public class FlutterAiuiPlugin implements FlutterPlugin, MethodCallHandler, ActivityAware {
    private MethodChannel channel;
    private AIUIWrapper aiuiWrapper;
    private Context context;
    private Activity activity;

    @Override
    public void onAttachedToEngine(@NonNull FlutterPluginBinding flutterPluginBinding) {
        context = flutterPluginBinding.getApplicationContext();
        channel = new MethodChannel(flutterPluginBinding.getBinaryMessenger(), "flutter_aiui");
        channel.setMethodCallHandler(this);
    }

    @Override
    public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) {
        switch (call.method) {
            case Constant.INIT:
                if (aiuiWrapper == null) {
                    aiuiWrapper = new AIUIWrapper(context, activity, channel, result, call);
                } else {
                    aiuiWrapper.initAIUIAgent(call.argument("config"), result);
                }
                break;
            case Constant.START_SPEAK:
                aiuiWrapper.startSpeak(result, call.arguments());
                break;
            case Constant.END_SPEAK:
                aiuiWrapper.endSpeak();
                result.success(null);
                break;
            case Constant.PAUSE_SPEAK:
                aiuiWrapper.onPause();
                result.success(null);
                break;
            case Constant.RESUME_SPEAK:
                aiuiWrapper.onResume(result, call.arguments());
                result.success(null);
                break;
            case Constant.WRITE_TEXT: {
                if (call.argument("text") == null) {
                    result.error(null, "文字不能为空", null);
                } else {
                    aiuiWrapper.writeText(Objects.requireNonNull(call.argument("text")),
                            call.argument("config"));
                    result.success(null);
                }
            }
            break;
            case Constant.DISPOSE:
                aiuiWrapper.destroyAgent(result);
                result.success(null);
                break;
            case Constant.SET_PARAM:
                aiuiWrapper.setParameters(call.arguments());
                result.success(null);
                break;
            case Constant.START_TTS:
                aiuiWrapper.startTTs(call.argument("text"), call.argument("config"),
                        Objects.equals(call.argument("resume"), true));
                result.success(null);
                break;
            case Constant.STOP_TTS:
                aiuiWrapper.stopTTS();
                result.success(null);
                break;
            case Constant.PAUSE_TTS:
                aiuiWrapper.pauseTTS();
                result.success(null);
                break;
            default:
                result.notImplemented();
                break;
        }

    }

    @Override
    public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
        channel.setMethodCallHandler(null);
    }

    @Override
    public void onAttachedToActivity(@NonNull ActivityPluginBinding binding) {
        activity = binding.getActivity();
    }

    @Override
    public void onDetachedFromActivityForConfigChanges() {

    }

    @Override
    public void onReattachedToActivityForConfigChanges(@NonNull ActivityPluginBinding binding) {
        onAttachedToActivity(binding);
    }

    @Override
    public void onDetachedFromActivity() {

    }
}

File: android/src/main/kotlin/com/tianxin/flutter_aiui/FlutterAiuiPlugin.kt
package com.tianxin.flutter_aiui

import android.Manifest
import android.app.Activity
import android.content.Context
import android.os.Build
import androidx.core.app.ActivityCompat
import com.iflytek.aiui.*
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.embedding.engine.plugins.activity.ActivityAware
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result
import java.util.*
import java.util.concurrent.ScheduledExecutorService
import java.util.concurrent.ScheduledThreadPoolExecutor

/** FlutterAiuiPlugin */
class FlutterAiuiPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
    private var aiuiWrapper: AIUIWrapper? = null
    private lateinit var context: Context
    private lateinit var activity: Activity
    private lateinit var channel: MethodChannel
    private lateinit var scheduledExecutorService: ScheduledExecutorService

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        context = flutterPluginBinding.applicationContext
        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "flutter_aiui")
        channel.setMethodCallHandler(this)
        scheduledExecutorService = ScheduledThreadPoolExecutor(1)
    }

    @Suppress("UNCHECKED_CAST")
    override fun onMethodCall(call: MethodCall, result: Result) {
        if (call.method != Constant.INIT_AGENT && aiuiWrapper == null) {
            result.error(Constant.AGENT_NOT_CREATE, "AiuiWrapper未初始化", null)
            return
        }
        when (call.method) {
            Constant.INIT_AGENT -> {
                if (aiuiWrapper == null) {
                    aiuiWrapper = AIUIWrapper(
                        context = context,
                        channel = channel,
                        executorService = scheduledExecutorService,
                    )
                }
                aiuiWrapper!!.initAgent(call.arguments as String?, result)
            }
            Constant.DESTROY_AGENT -> aiuiWrapper!!.destroyAgent(result)
            Constant.SET_PARAMS -> aiuiWrapper!!.setParams(
                call.arguments as String,
                result,
            )
            Constant.START_RECORD_AUDIO -> aiuiWrapper!!.startRecordAudio(
                call.arguments as String?,
                result
            )
            Constant.PAUSE_RECORD_AUDIO -> aiuiWrapper!!.pauseRecordAudio(
                call.arguments as String?,
                result
            )
            Constant.RESUME_RECORD_AUDIO -> aiuiWrapper!!.resumeRecordAudio(
                call.arguments as String?,
                result
            )
            Constant.STOP_RECORD_AUDIO -> aiuiWrapper!!.stopRecordAudio(
                call.arguments as String?,
                result
            )
            Constant.WRITE_TEXT -> aiuiWrapper!!.writeText(
                call.arguments as HashMap<String, Any>,
                result
            )
            Constant.START_TTS -> aiuiWrapper!!.startTTS(
                call.arguments as HashMap<String, Any>,
                false,
                result
            )
            Constant.PAUSE_TTS -> aiuiWrapper!!.pauseTTS(result)
            Constant.RESUME_TTS -> aiuiWrapper!!.startTTS(null, true, result)
            Constant.STOP_TTS -> aiuiWrapper!!.stopTTS(result)
            else -> result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }

    override fun onAttachedToActivity(binding: ActivityPluginBinding) {
        activity = binding.activity
        requestPermissions()
    }

    override fun onDetachedFromActivityForConfigChanges() {
    }

    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
        onAttachedToActivity(binding)
    }

    override fun onDetachedFromActivity() {
    }

    private fun requestPermissions() {
        try {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
                val permissions = arrayOf(
                    Manifest.permission.READ_PHONE_STATE,
                    Manifest.permission.RECORD_AUDIO,
                    Manifest.permission.WRITE_EXTERNAL_STORAGE,
                    Manifest.permission.ACCESS_FINE_LOCATION,
                    Manifest.permission.INTERNET
                )
                ActivityCompat.requestPermissions(activity, permissions, 0x0010
                )
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}
